### ISSUE TITLE
Serverside failure to restore wallet can trap user in error dialog

### ISSUE THREAD

**Issue by sneurlax** [cypherstack/stack_wallet#749]
You can reproduce this by getting banned from the server for creating too many wallets (like by creating 20+ eCash wallets).

This patch allows us to escape the dialog, but pollutes our wallets so the wallets dialog view throws:
```
Index: lib/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart b/lib/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart
--- a/lib/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart	(revision 5f9692a55751081d4d3324d9987acaace5e363d4)
+++ b/lib/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart	(date 1707351408487)
@@ -13,6 +13,7 @@
 import 'package:stackwallet/providers/global/secure_store_provider.dart';
 import 'package:stackwallet/providers/providers.dart';
 import 'package:stackwallet/themes/stack_colors.dart';
+import 'package:stackwallet/utilities/logger.dart';
 import 'package:stackwallet/utilities/text_styles.dart';
 import 'package:stackwallet/wallets/isar/providers/wallet_info_provider.dart';
 import 'package:stackwallet/widgets/stack_dialog.dart';
@@ -65,10 +66,23 @@
             style: STextStyles.itemSubtitle12(context),
           ),
           onPressed: () async {
-            await ref.read(pWallets).deleteWallet(
-                  ref.read(pWalletInfo(walletId)),
-                  ref.read(secureStoreProvider),
-                );
+            try {
+              // If we fail to restore, we can get stuck in this dialog.
+              //
+              // This try-catch at least gets us out of the dialog.
+              // But it breaks the wallets dialog view, so I'm not committing it.
+              // TODO [prio=low]: Ensure that the incomplete wallet is cleaned up.
+              await ref.read(pWallets).deleteWallet(
+                    ref.read(pWalletInfo(walletId)),
+                    ref.read(secureStoreProvider),
+                  );
+            } catch (e, s) {
+              Logging.instance.log(
+                "Failed to delete wallet after restore failed"
+                "\nError: $e\nStack trace: $s",
+                level: LogLevel.Error,
+              );
+            }
 
             if (mounted) {
               Navigator.of(context).pop();
```

**Comment by sneurlax** [comment]
```
#0      ProviderElementBase.requireState.<anonymous closure> (package:riverpod/src/framework/provider_base.dart:360:25)
#1      ResultError.map (package:riverpod/src/framework/provider_base.dart:1110:17)
#2      ProviderElementBase.requireState (package:riverpod/src/framework/provider_base.dart:359:18)
#3      ProviderElementBase.readSelf (package:riverpod/src/framework/provider_base.dart:808:12)
#4      ProviderContainer.read (package:riverpod/src/framework/container.dart:300:20)
#5      ConsumerStatefulElement.read (package:flutter_riverpod/src/consumer.dart:487:59)
#6      _RestoreFailedDialogState.build.<anonymous closure> (package:stackwallet/pages/add_wallet_views/restore_wallet_view/sub_widgets/restore_failed_dialog.dart:69:23)
#7      _InkResponseState.handleTap (package:flutter/src/material/ink_well.dart:1183:21)
#8      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:275:24)
#9      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:652:11)
#10     BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:309:5)
#11     BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:279:7)
#12     GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:167:27)
#13     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:492:20)
#14     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:468:22)
#15     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:439:11)
#16     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:413:7)
#17     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:376:5)
#18     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:323:7)
#19     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:292:9)
#20     _invoke1 (dart:ui/hooks.dart:328:13)
#21     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:410:7)
#22     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)

flutter: didChangeAppLifecycleState: inactive
```

**Comment by sneurlax** [comment]
Closed by #764
